compiler/ballerina-parser/src/test/resources/misc/ambiguity/ambiguity_assert_29.json

```diff
@@ -74,169 +74,113 @@
                   "hasDiagnostics": true,
                   "children": [
                     {
-                      "kind": "ACTION_STATEMENT",
```

Comment:
Now this action statement has affected.  `(a public| b) => a + b -> w;` Not sure why.

Reply:
Had an offline discussion. The changes are expected. Hence resolving the related comments.

Reply:
I'm having second thoughts on this. 
https://github.com/ballerina-platform/ballerina-lang/blob/8aa1e0cdabf9672d90d1eee73171d991d706fe40/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L11175-L11177

Here have used the Query precedence to represent the lowest expression precedence.  Now we have to use the new expression lowest expression precedence which is `ANON_FUNC_OR_LET`. This is because of the right associativity in `(x) => y `

The above should parse as `((a public| b) => a + b) -> w;`

Reply:
Addition to above plave, we have two other places also to change,
1. https://github.com/ballerina-platform/ballerina-lang/blob/8aa1e0cdabf9672d90d1eee73171d991d706fe40/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L10616-L10618
2. https://github.com/ballerina-platform/ballerina-lang/blob/8aa1e0cdabf9672d90d1eee73171d991d706fe40/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L11224-L11226